feat: new Shell (jump to) home screen

caybro opened the pull request:
### What does the PR do

- Shell becomes the new startup/home screen; hidden behind a feature flag
- has a searchable grid, providing an overview of chats, communities, wallets, settings etc
- some related smaller component and storybook page fixes
- cleanup some clazy warnings

(additional fixes in separate commits)

Fixes [#18032](https://github.com/status-im/status-desktop/issues/18032)
Part of https://github.com/status-im/status-desktop/issues/17971

### Affected areas

AppMain, Shell

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://github.com/user-attachments/assets/1e2b5862-1cbc-4f9f-ac1d-dca523700199

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | e7070cb2 | [#2](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-17916/2/) | 2025-05-12 13:58:39 | ~9 min | `tests/nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-17916/2//consoleText) |

caybro commented on ui/app/AppLayouts/Shell/ShellAdaptor.qml:
```diff
@@ -0,0 +1,300 @@
+
+    SortFilterProxyModel {
+        id: filteredCombinedModel
+        sourceModel: shellProxyModel
```
That's a writable overlay :) Adding the "timestamp" (sorting) and "pinned" properties

caybro replied:
... and it's gone :) Replaced with a writable ObjectProxyModel and some JS code

caybro commented on ui/app/AppLayouts/Shell/ShellContainer.qml:
```diff
@@ -0,0 +1,214 @@
+            onClicked: root.notificationButtonClicked()
+        }
+
+        // TODO make a separate/shared component
```
Ah right... ypu I think we should make this one a reusable component, since we use it elsewhere in the app too

jrainville replied:
I think it would be nice yeah. I assume this is basically a duplicate of the one in the bottom left corner (except smaller)?

caybro replied:
Done, extracted into a separate component

caybro commented on ui/app/AppLayouts/Shell/ShellDockButton.qml:
```diff
@@ -0,0 +1,136 @@
+ToolButton {
+    id: root
+    
+    /*required */property int sectionType // cf Constants.appSection.*
```
It should be `required` but it crashes in Qt5 :(

caybro commented on ui/app/AppLayouts/Shell/ShellDockButton.qml:
```diff
@@ -0,0 +1,136 @@
+    icon.height: 36
+
+    background: Rectangle {
+        color: Qt.rgba(1, 1, 1, hovered ? 0.1 : 0.05) // FIXME get rid of opacity tricks
```
No, not really, it's that the designers like to play with opacities (which has a lot of nasty side effects) rather than defining the colors flat

caybro commented on ui/app/AppLayouts/Shell/delegates/ShellGridChatItem.qml:
```diff
@@ -0,0 +1,46 @@
+        }
+    }
+
+    // TODO bottomRowComponent -> last message in this chat
```
Ah right... I meant to ask you: can we get that easily somehow from the chat section? :)

jrainville replied:
It's available in the status-go return. We sued to show the lastMessage in the chat list on the left, but we stopped when we changed the design.

So either it's still in the model or it's a matter of adding it back.

TLDR: yes it's doable

caybro commented on ui/app/AppLayouts/Shell/delegates/ShellGridCommunityItem.qml:
```diff
@@ -0,0 +1,97 @@
+    Component {
+        id: pendingTagComponent
+        CustomInfoTag {
+            asset.name: "history" // TODO correct new "pending" icon
```
Not until we import the "new" icons for the whole app

caybro commented on ui/app/mainui/AppMain.qml:
```diff
@@ -1095,6 +1114,8 @@ Item {
 
         function openSearchPopup() {
+            if (shellLoader.active)
+                return
```
Kinda clashing with this search here... not sure, will ask Ben

caybro commented on ui/app/mainui/AppMain.qml:
```diff
@@ -2352,6 +2385,88 @@ Item {
+                readonly property bool sectionsLoaded: appMain.rootStore.mainModuleInst && appMain.rootStore.mainModuleInst.sectionsLoaded
+
+                sectionsBaseModel: sectionsLoaded ? appMain.rootStore.mainModuleInst.sectionsModel : null
+                chatsBaseModel: sectionsLoaded ? appMain.rootStore.mainModuleInst.getChatSectionModule().model // FIXME model null initially or with empty images
```
Yeah I think there's some missing "dataChanged" signal somewhere in NIM code; will open a separate issue

caybro commented on ui/app/mainui/AppMain.qml:
```diff
@@ -2352,6 +2385,88 @@ Item {
+                    if (itemId == Constants.settingsSubsection.backUpSeed) {
+                        return Global.openBackUpSeedPopup()
+                    } else if (itemId == Constants.settingsSubsection.signout) {
+                        return Qt.exit(0) // TODO confirm dialog?
```
Yup, good point; will add it (make it shared between here and the Settings)

caybro commented on src/app/modules/shared_models/section_model.nim:
```diff
@@ -309,7 +309,6 @@ QtObject:
     updateRoleWithValue(joined, Joined, item.joined)
     updateRoleWithValue(spectated, Spectated, item.spectated)
     updateRoleWithValue(isMember, IsMember, item.isMember)
-    updateRoleWithValue(isMember, IsMember, item.isMember)
```
KK, I can fix that one

caybro commented:
@jrainville addressed your comments/feedback (in separate followup commits); pls have a second look. I can raise separate issues for the missing/faulty chat section behavior

jrainville commented:
I just realized, in your video, the chat icons are square for the ones without images: 
![image](https://github.com/user-attachments/assets/517ac299-1cf6-4409-826b-5509a2218bf5)

caybro commented:
> I just realized, in your video, the chat icons are square for the ones without images: 
> ![image](https://github.com/user-attachments/assets/517ac299-1cf6-4409-826b-5509a2218bf5)
> 

Yeah I fixed that already in the meantime, just forgot to update the video

jrainville reviewed (approved):
Nice work. Code looks good to me!

Can you create the issues for the 2-3 points still in comments. You can put them as children of the main issue.

I can work on them Monday

noeliaSD commented on ui/app/AppLayouts/Shell/delegates/ShellGridItem.qml:
```diff
@@ -0,0 +1,198 @@
+    property string subtitle: Utils.translatedSectionName(sectionType)
+    property color color
+    property url banner
+    property bool hasNotification
```
```suggestion
    readonly property bool hasNotification: root. notificationsCount > 0
```

caybro replied:
No, they are different things

noeliaSD commented on ui/app/AppLayouts/Shell/delegates/ShellGridItem.qml:
```diff
@@ -0,0 +1,198 @@
+            anchors.bottom: parent.bottom
+            width: parent.width
+
+            color: hovered ? "#222833" : "#161c27"
```
`Theme` or at least `d` readonly property

noeliaSD commented on ui/app/AppLayouts/Shell/delegates/ShellGridWalletItem.qml:
```diff
@@ -0,0 +1,59 @@
+    subtitle: SQUtils.Utils.elideAndFormatWalletAddress(root.itemId)
+
+    background: Rectangle {
+        color: hovered ? "#222833" : "#161c27"
```
Same

noeliaSD commented on ui/app/mainui/AppMain.qml:
```diff
@@ -954,11 +962,19 @@ Item {
             if (sectionType === Constants.appSection.profile) {
-                profileLoader.settingsSubsection = subsection
-                profileLoader.settingsSubSubsection = subSubsection;
+                profileLoader.settingsSubsection = subsection || Constants.settingsSubsection.profile
```
Coul you explain why this `|| Constants.settingsSubsection.profile`?

caybro replied:
That's the default Settings subsection (Profile), if not provided

noeliaSD reviewed (commented):
Good job! I've added some comments / questions.

And here some general design / usability comments / suggestions after testing:
- It's a bit annoying to me that the grid order changes each time I click a new item. It's nice to have the most recents on top but then it makes it difficult to find a specific one since it's never at the same position. Maybe we could have the possibility of ordering by recents or not, so user can decide. Maybe it's more a question for the design team @benjthayer, wdyt? 
- I'd make the cards wider so we have less in a row. I find it a bit overwhelming to have so much information on the same initial page.
- Why do we have it just in "dark mode"?
- Would it make sense to add the `Sign out & Quit` option in this view too?

caybro commented on storybook/pages/ShellContainerPage.qml:
```diff
@@ -0,0 +1,188 @@
+            showDapps: ctrlShowDapps.checked || ctrlShowAllEntries.checked
+
+            showEnabledSectionsOnly: ctrlShowEnabledSectionsOnly.checked
+            marketEnabled: ctrlMarketEnabled.checked
```
Yeah this (orthogonal) logic is coming from AppMain.qml:
```qml
ValueFilter {
    roleName: "sectionType"
    value: Constants.appSection.swap
    enabled: !appMain.featureFlagsStore.marketEnabled
}
ValueFilter {
    roleName: "sectionType"
    value: Constants.appSection.market
    enabled: appMain.featureFlagsStore.marketEnabled
}
```
Market is preferred (if enabled), otherwise we get the Swap button

caybro commented on ui/app/AppLayouts/Shell/ShellContainer.qml:
```diff
@@ -0,0 +1,165 @@
+            getEmojiHashFn: root.getEmojiHashFn
+            getLinkToProfileFn: root.getLinkToProfileFn
+
+            badge.border.color: hovered ? "#222833" : "#161c27"
```
Nope, all the colors here are very specific

caybro commented on ui/StatusQ/src/assets/img/icons/shell/communities.svg:
A different one

caybro commented on ui/StatusQ/src/assets/img/icons/shell/wallet.svg:
A different one

caybro commented on ui/app/mainui/AppMain.qml:
```diff
@@ -954,11 +962,19 @@ Item {
             } else if (sectionType === Constants.appSection.wallet) {
                 appView.children[Constants.appViewStackIndex.wallet].item.openDesiredView(subsection, subSubsection, data)
+            } else if (sectionType === Constants.appSection.swap) {
+                d.launchSwap()
```
Ah, thanks! That looks like a bad rebase from my side :grimacing:

caybro replied:
Fixed https://github.com/status-im/status-desktop/pull/17916/commits/8d30d2816ffd05b0fc62ad829c1e65ea6fd07ec2

micieslak commented on ui/StatusQ/src/shellproxymodel.cpp:
```diff
@@ -0,0 +1,234 @@
+#include "StatusQ/shellproxymodel.h"
```
I think that we should add generic model to our toolbox instead of creating case-specific model there. It could be kind of "lightweight" version of WritableProxyModel, where we can define additional writable roles.

There is a chance it can be done on top of existing internal utility `ModelSyncedContainer` allowing to store some additional metadata synced with the model but with O(1) read/write.

micieslak replied:
Another idea is to use one more `ObjectProxyModel` for now, in very simple form:

```qml
    ObjectProxyModel {
        delegate: QtObject {
            property real timestamp: 0
            property bool pinned: false
        }

        exposedRoles: ["timestamp", "pinned"]

        sourceModel: srcModel
    }
```
Basically it adds two additional writable roles. It should work nicely with separated loading/saving.

caybro replied:
OK, did it as you said :) Custom C++ model dropped completely

micieslak reviewed (commented):
It's great work! I really like the nice separation of UI and data transformation and proxy models in action allowing to fit it into exising app without too many changes! :muscle: 

I publish my review regarding the shell itself, later I will take a closer look at the integration part and maybe leave some comment there as well if found sth.

Thanks!

caybro commented on ui/StatusQ/include/StatusQ/shellproxymodel.h:
```diff
@@ -0,0 +1,60 @@
+    Q_PROPERTY(QString profileId READ profileId WRITE setProfileId NOTIFY profileIdChanged REQUIRED FINAL)
+
+public:
+    enum ExtraRoles {
+        TimestampRole = Qt::UserRole + 1,
+        PinnedRole
+    };
```
Right, I'll increase the starting point

caybro replied:
No longer applies

caybro commented on ui/StatusQ/src/modelentry.cpp:
```diff
@@ -111,7 +111,7 @@ void ModelEntry::setSourceModel(QAbstractItemModel* sourceModel)
             &QAbstractItemModel::dataChanged,
             this,
-            [this](const QModelIndex& topLeft, const QModelIndex& bottomRight, const QVector<int>& roles) {
+            [this](const QModelIndex& topLeft, const QModelIndex& bottomRight, const auto& roles) {
```
I just thought it nicer since the signature changed between versions; and clazy was complaining here on Qt 6; thix `auto` syntax works for both (Qt5 and Qt6)

anastasiyaig commented on ui/app/AppLayouts/Shell/ShellDockButton.qml:
```diff
@@ -0,0 +1,136 @@
+
+import utils 1.0
+
+ToolButton {
```
can we add object name for button delegate? Something like ShellButton_Market , etc?

caybro commented on ui/app/AppLayouts/Shell/ShellDock.qml:
```diff
@@ -0,0 +1,123 @@
+import utils 1.0
+
+Control {
+    id: root
```
Sure, will add some objectNames for the toplevel components

caybro replied:
Added some object names in a separate commit: https://github.com/status-im/status-desktop/pull/17916/commits/3fb9f3c0d222081c03d55e90caec73b0589faa61

Just say so and I'll add more if needed :)

caybro commented on ui/app/AppLayouts/Shell/ShellDock.qml:
```diff
@@ -0,0 +1,123 @@
+    id: root
+
+    required property var sectionsModel
+    required property var pinnedModel
```
Basically the same as above but yeah... will add :)

caybro commented on ui/StatusQ/src/modelentry.cpp:
```diff
@@ -328,7 +328,7 @@ void ModelEntry::notifyItemChanges(const QStringList& roles)
     }
 
-    for(auto role : roles)
+    for(const auto& role : roles)
```
`role` is a `QString` here so I think clazy was right in complaining here

caybro commented on ui/app/AppLayouts/Shell/ShellDock.qml:
```diff
@@ -0,0 +1,123 @@
+Control {
+    id: root
+
+    required property var sectionsModel
```
Fixed in https://github.com/status-im/status-desktop/pull/17916/commits/799ab74003f0b32529f99d02fd1247c5ee671fee

caybro commented on ui/app/AppLayouts/Shell/ShellGrid.qml:
```diff
@@ -0,0 +1,171 @@
+    }
+
+    Component {
+        id: communityDelegate
```
Fixed

caybro commented on ui/StatusQ/src/shellproxymodel.cpp:
```diff
@@ -0,0 +1,234 @@
+    endResetModel();
+}
+
+void ShellProxyModel::save()
```
Yup, fixed too

micieslak commented on ui/app/AppLayouts/Shell/ShellAdaptor.qml:
```diff
@@ -0,0 +1,401 @@
+        }
+    }
+
+    function save() {
```
instead saving only onDestruciton we could also use `Timer` reseted by successful `setPinned` or `setTimestamp`. Then there is no risk of loosing the arrangement e.g. in case of app crash

caybro replied:
I tried forcefully killing the app and it still saved the entries :)

micieslak commented on ui/app/AppLayouts/Shell/delegates/ShellGridCommunityItem.qml:
```diff
@@ -0,0 +1,97 @@
+    sectionType: Constants.appSection.community
+
+    function numberFormat(number) {
+        var res = number
```
```suggestion
        let res = number
```

caybro replied:
Why and how is this better? :)

micieslak commented on ui/StatusQ/src/objectproxymodel.cpp:
```diff
@@ -230,7 +229,7 @@ void ObjectProxyModel::emitAllDataChanged()
         return;
 
-    if (m_expectedRoles.isEmpty())
+    if (m_exposedRolesSet.isEmpty())
```
nice find :sweat_smile:

micieslak commented on ui/StatusQ/src/snapshotmodel.cpp:
```diff
@@ -48,7 +48,7 @@ void SnapshotModel::grabSnapshot(const QAbstractItemModel& model, bool recursive
     m_roles = model.roleNames();
 
-    auto roles = m_roles.keys();
+    const auto roles = m_roles.keys();
```
hmm, using `std::as_const` to call const version of `keys` is valid point here, right?

micieslak commented on ui/app/AppLayouts/Shell/ShellAdaptor.qml:
```diff
@@ -0,0 +1,401 @@
+    }
+
+    function load() {
+        var length = 0
```
```suggestion
        let length = 0
```

micieslak commented on ui/app/AppLayouts/Shell/ShellAdaptor.qml:
```diff
@@ -0,0 +1,401 @@
+    function load() {
+        var length = 0
+        const settingsData = shellSettings.value("ShellEntries")
+        var dataArray = []
```
```suggestion
        let dataArray = []
```

caybro replied:
Fixed

micieslak commented on ui/app/AppLayouts/Shell/ShellAdaptor.qml:
```diff
@@ -0,0 +1,401 @@
+            dataArray = JSON.parse(settingsData)
+            length = dataArray.length
+        } catch (e) {
+            return
```
maybe some warning here would be useful here

micieslak commented on ui/app/AppLayouts/Shell/ShellAdaptor.qml:
```diff
@@ -0,0 +1,401 @@
+            return
+        }
+
+        dataArray.forEach(function(item) {
```
nitpick:
```suggestion
        dataArray.forEach(item =>  {
```

caybro replied:
Yeah... but it just messes up the indentation, in fact that's the only reason I use the longer `function()` syntax

micieslak commented on ui/app/AppLayouts/Shell/ShellAdaptor.qml:
```diff
@@ -0,0 +1,401 @@
+        }
+
+        dataArray.forEach(function(item) {
+            let idx = ModelUtils.indexOf(shellProxyModel, "key", item.key)
```
```suggestion
            const idx = ModelUtils.indexOf(shellProxyModel, "key", item.key)
```

caybro replied:
Fixed

micieslak commented on ui/app/AppLayouts/Shell/ShellContainer.qml:
```diff
@@ -0,0 +1,165 @@
+Control {
+    id: root
+
+    required property ShellAdaptor shellAdaptor
```
so, are we proceeding with that or leaving as it is?

micieslak commented on ui/app/AppLayouts/Shell/ShellContainer.qml:
```diff
@@ -0,0 +1,179 @@
+            leftMargin: d.isNarrowView ? 0 : root.width * .1
+            rightMargin: d.isNarrowView ? 0 : root.width * .1
+
+            onItemActivated: function(key, sectionType, itemId) {
```
what about using shorter notation wherever possible?
```suggestion
            onItemActivated: (key, sectionType, itemId) => {
```

micieslak commented on storybook/pages/ShellContainerPage.qml:
On Qt 5.15.2 it reports bunch of warnings when launched and used:
```
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellAdaptor.qml:238:13: Unable to assign [undefined] to QString
file:///home/michalc/workspace/status-desktop/ui/imports/utils/Utils.qml:934: TypeError: Cannot call method 'isCompressedPubKey' of null
file:///home/michalc/Qt/5.15.2/gcc_64/qml/QtQuick/Controls.2/Universal/ScrollBar.qml:82:13: QML NumberAnimation: Cannot animate non-existent property "opacity"
file:///home/michalc/Qt/5.15.2/gcc_64/qml/QtQuick/Controls.2/Universal/ScrollBar.qml:87:17: QML PropertyAction: Cannot animate non-existent property "opacity"
file:///home/michalc/Qt/5.15.2/gcc_64/qml/QtQuick/Controls.2/Universal/ScrollBar.qml:89:17: QML NumberAnimation: Cannot animate non-existent property "opacity"
file:///home/michalc/Qt/5.15.2/gcc_64/qml/QtQuick/Controls.2/Universal/ScrollBar.qml:82:13: QML NumberAnimation: Cannot animate non-existent property "opacity"
file:///home/michalc/Qt/5.15.2/gcc_64/qml/QtQuick/Controls.2/Universal/ScrollBar.qml:87:17: QML PropertyAction: Cannot animate non-existent property "opacity"
file:///home/michalc/Qt/5.15.2/gcc_64/qml/QtQuick/Controls.2/Universal/ScrollBar.qml:89:17: QML NumberAnimation: Cannot animate non-existent property "opacity"
file:///home/michalc/Qt/5.15.2/gcc_64/qml/QtQuick/Controls.2/Universal/ScrollBar.qml:82:13: QML NumberAnimation: Cannot animate non-existent property "opacity"
file:///home/michalc/Qt/5.15.2/gcc_64/qml/QtQuick/Controls.2/Universal/ScrollBar.qml:87:17: QML PropertyAction: Cannot animate non-existent property "opacity"
file:///home/michalc/Qt/5.15.2/gcc_64/qml/QtQuick/Controls.2/Universal/ScrollBar.qml:89:17: QML NumberAnimation: Cannot animate non-existent property "opacity"
qml: !!! ITEM ACTIVATED; key:  ; sectionType: 1 ; itemId: 
qml: !!! ITEM ACTIVATED; key:  ; sectionType: 3 ; itemId: 
qml: !!! ITEM ACTIVATED; key: 2;id13 ; sectionType: 2 ; itemId: id13
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:140:13: Unable to assign [undefined] to bool
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:140:13: Unable to assign [undefined] to bool
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:194:13: Unable to assign [undefined] to QUrl
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:123:13: Unable to assign [undefined] to bool
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:119:13: Unable to assign [undefined] to int
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:114:13: Unable to assign [undefined] to QUrl
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:122:13: Unable to assign [undefined] to bool
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:120:13: Unable to assign [undefined] to int
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Components/StatusRoundIcon.qml:26:5: QML StatusIcon: Cannot open: file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/dapp
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:116:13: Unable to assign [undefined] to int
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:115:13: Unable to assign [undefined] to bool
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:140:13: Unable to assign [undefined] to bool
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:158:13: Unable to assign [undefined] to QString
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:159:13: Unable to assign [undefined] to QString
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Components/StatusLetterIdenticon.qml:39:5: QML StatusEmoji: Cannot open: file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/assets/twemoji/svg/75.svg
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:123:13: Unable to assign [undefined] to bool
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:119:13: Unable to assign [undefined] to int
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:114:13: Unable to assign [undefined] to QUrl
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:122:13: Unable to assign [undefined] to bool
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:120:13: Unable to assign [undefined] to int
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Components/StatusRoundIcon.qml:26:5: QML StatusIcon: Cannot open: file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/☀️
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:140:13: Unable to assign [undefined] to bool
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:177:13: Unable to assign [undefined] to int
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:178:13: Unable to assign [undefined] to int
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:123:13: Unable to assign [undefined] to bool
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:119:13: Unable to assign [undefined] to int
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:114:13: Unable to assign [undefined] to QUrl
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:122:13: Unable to assign [undefined] to bool
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:120:13: Unable to assign [undefined] to int
file:///home/michalc/Qt/5.15.2/gcc_64/qml/QtQuick/Controls.2/Universal/ScrollBar.qml:82:13: QML NumberAnimation: Cannot animate non-existent property "opacity"
file:///home/michalc/Qt/5.15.2/gcc_64/qml/QtQuick/Controls.2/Universal/ScrollBar.qml:87:17: QML PropertyAction: Cannot animate non-existent property "opacity"
file:///home/michalc/Qt/5.15.2/gcc_64/qml/QtQuick/Controls.2/Universal/ScrollBar.qml:89:17: QML NumberAnimation: Cannot animate non-existent property "opacity"
qml: !!! ITEM ACTIVATED; key: 1;id1 ; sectionType: 1 ; itemId: id1
```

micieslak replied:
And on Qt 6.9:
```
file:///home/michalc/workspace/status-desktop/storybook/main.qml:337:5: QML Settings: The Settings type from Qt.labs.settings is deprecated and will be removed in a future release. Please use the one from QtCore instead.
file:///home/michalc/workspace/status-desktop/storybook/src/Storybook/LogsAndControlsPanel.qml:46:5: QML Settings: The Settings type from Qt.labs.settings is deprecated and will be removed in a future release. Please use the one from QtCore instead.
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellAdaptor.qml:322:5: QML Settings: The Settings type from Qt.labs.settings is deprecated and will be removed in a future release. Please use the one from QtCore instead.
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Core/LocaleUtils.qml:290:37: QML Settings: The Settings type from Qt.labs.settings is deprecated and will be removed in a future release. Please use the one from QtCore instead.
file:///home/michalc/workspace/status-desktop/ui/imports/utils/Utils.qml:934: TypeError: Cannot call method 'isCompressedPubKey' of null
Storybook started, Qt runtime version: 6.9.0 ; built against version: "6.9.0" installed in: "/home/michalc/Qt/6.9.0/gcc_64" ; QQC style: "Universal"
qt.qml.defaultmethod: Assigning an object to a signal handler is deprecated. Instead, create the object, give it an id, and call the desired slot from the signal handler. The object is: QQuickSequentialAnimation(0x64e4352d3990)
qt.qml.defaultmethod: Assigning an object to a signal handler is deprecated. Instead, create the object, give it an id, and call the desired slot from the signal handler. The object is: QQuickSequentialAnimation(0x64e4353cc8a0)
qt.qml.defaultmethod: Assigning an object to a signal handler is deprecated. Instead, create the object, give it an id, and call the desired slot from the signal handler. The object is: QQuickSequentialAnimation(0x64e433d2f450)
qt.qml.defaultmethod: Assigning an object to a signal handler is deprecated. Instead, create the object, give it an id, and call the desired slot from the signal handler. The object is: QQuickSequentialAnimation(0x64e435429790)
qt.qml.defaultmethod: Assigning an object to a signal handler is deprecated. Instead, create the object, give it an id, and call the desired slot from the signal handler. The object is: QQuickSequentialAnimation(0x64e435448230)
qt.qml.defaultmethod: Assigning an object to a signal handler is deprecated. Instead, create the object, give it an id, and call the desired slot from the signal handler. The object is: QQuickSequentialAnimation(0x64e435a726f0)
qml: !!! ITEM ACTIVATED; key: 1;id5 ; sectionType: 1 ; itemId: id5
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:177:13: Unable to assign [undefined] to int
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Components/StatusLetterIdenticon.qml:39:5: QML StatusEmoji: Cannot open: file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/assets/twemoji/svg/68.svg
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Components/StatusLetterIdenticon.qml:39:5: QML StatusEmoji: Cannot open: file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/assets/twemoji/svg/68.svg
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Components/StatusLetterIdenticon.qml:39:5: QML StatusEmoji: Cannot open: file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/assets/twemoji/svg/68.svg
qrc:/qt-project.org/imports/QtQuick/Controls/Universal/ScrollBar.qml:55:17: QML PropertyAction: Cannot animate non-existent property "opacity"
qml: !!! ITEM 2;id15 PINNED: true
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:177:13: Unable to assign [undefined] to int
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:177:13: Unable to assign [undefined] to int
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:123:13: Unable to assign [undefined] to bool
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:114:13: Unable to assign [undefined] to QUrl
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:123:13: Unable to assign [undefined] to bool
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:114:13: Unable to assign [undefined] to QUrl
qt.qml.defaultmethod: Assigning an object to a signal handler is deprecated. Instead, create the object, give it an id, and call the desired slot from the signal handler. The object is: QQuickSequentialAnimation(0x64e434d28c70)
qrc:/qt-project.org/imports/QtQuick/Controls/Universal/ScrollBar.qml:55:17: QML PropertyAction: Cannot animate non-existent property "opacity"
qml: !!! ITEM 1;id3 PINNED: true
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:123:13: Unable to assign [undefined] to bool
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:114:13: Unable to assign [undefined] to QUrl
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:177:13: Unable to assign [undefined] to int
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Shell/ShellGrid.qml:177:13: Unable to assign [undefined] to int
qt.qml.defaultmethod: Assigning an object to a signal handler is deprecated. Instead, create the object, give it an id, and call the desired slot from the signal handler. The object is: QQuickSequentialAnimation(0x64e435668a90)
```

micieslak reviewed (commented):
Nice! However, I've found a few minor issues probably worth addressing

caybro commented on ui/app/AppLayouts/Shell/ShellContainer.qml:
```diff
@@ -0,0 +1,185 @@
+
+    // dock (see ShellAdaptor for docu)
+    required property var sectionsModel
+    required property var pinnedModel
```
@micieslak taking the flat models now; adaptor is constructed outside

micieslak reviewed (approved):
Solid work!
